gobblin-modules/gobblin-http/src/main/java/gobblin/converter/AvroHttpJoinConverter.java

```diff
@@ -0,0 +1,105 @@
+package gobblin.converter;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+
+import lombok.extern.slf4j.Slf4j;
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.DataConversionException;
+import gobblin.converter.HttpJoinConverter;
+import gobblin.converter.SchemaConversionException;
+import gobblin.http.HttpOperation;
+import gobblin.http.HttpRequestResponseRecord;
+
+
+/**
+ * A type of {@link HttpJoinConverter} with AVRO as input and output format
+ *
+ * Input:
+ *    User provided record
+ *
+ * Output:
+ *    User provided record plus http request & response record
+ */
+@Slf4j
+public abstract class AvroHttpJoinConverter<RQ, RP> extends HttpJoinConverter<Schema, Schema, GenericRecord, GenericRecord, RQ, RP> {
+  public static final String HTTP_REQUEST_RESPONSE = "HttpRequestResponse";
+
+  @Override
+  public Schema convertSchemaImpl(Schema inputSchema, WorkUnitState workUnitState)
+      throws SchemaConversionException {
+    List<Schema.Field> fields = Lists.newArrayList();
+    for (Schema.Field field : inputSchema.getFields()) {
+      Schema.Field newField = new Schema.Field(field.name(), field.schema(), field.doc(), field.defaultValue(), field.order());
+      fields.add(newField);
+    }
+
+    Schema.Field requestResponseField = new Schema.Field(HTTP_REQUEST_RESPONSE, HttpRequestResponseRecord.getClassSchema(), "http output schema contains request url and return result", null);
+    fields.add(requestResponseField);
+
+    Schema combinedSchema = Schema.createRecord(inputSchema.getName(), inputSchema.getDoc() + " (Http request and response are contained)", inputSchema.getNamespace(), false);
+    combinedSchema.setFields(fields);
+    return combinedSchema;
+  }
+
+  /**
+   *  Properties "gobblin.converter.http.keys" should be defined in the workUnit
+   */
+  @Override
+  public HttpOperation generateHttpOperation (GenericRecord inputRecord, WorkUnitState workUnitState) {
```

Comment:
It looks like a util method that can be put into `HttpUtils`. A more generic function signature would be `HttpOperation generateHttpOperation(GenericRecord inputRecord, State state)` because we may use the method in something like `HttpSource` where `JobState` might be used instead of `WorkUnitState`